test/lib/actions/RunAction.test.ts

```diff
@@ -0,0 +1,124 @@
+import {InputProcessor, InputProcessorImpl} from "../../../src/lib/InputProcessor";
+import {RuleFilterFactoryImpl} from "../../../src/lib/RuleFilterFactory";
+import {RunEngineOptionsFactory} from "../../../src/lib/EngineOptionsFactory";
+import {RunAction} from "../../../src/lib/actions/RunAction";
+import {FakeDisplay} from "../FakeDisplay";
+import {Logger} from "@salesforce/core";
+import {Inputs, PathlessRuleViolation, RuleResult} from "../../../src/types";
+import * as path from "path";
+import {initContainer} from '../../../src/ioc.config';
+import {expect} from "chai";
+import {Pmd7CommandInfo, PmdCommandInfo} from "../../../src/lib/pmd/PmdCommandInfo";
+import {Controller} from "../../../src/Controller";
+import {after} from "mocha";
+import {Results} from "../../../src/lib/output/Results";
+import {PMD6_VERSION, PMD7_VERSION} from "../../../src/Constants";
+import {FakeResultsProcessorFactory, RawResultsProcessor} from "./fakes";
+
+const codeFixturesDir = path.join(__dirname, '..', '..', 'code-fixtures');
+const pathToSomeTestClass = path.join(codeFixturesDir, 'apex', 'SomeTestClass.cls');
+const pathToCodeForCpd = path.join(codeFixturesDir, 'cpd');
+
+
+describe("Tests for RunAction", () => {
```

Comment:
These new tests are practically end to end tests except that they do not invoke the scanner - but the action directly. Also, I removed the output processing since that is all tested separetely.

Eventually I believe we will be moving most of our end to end tests to this format - leaving only a few actual end to end tests behind to confirm that we are wired up correctly to the cli and the output processors.

Reply:
This looks good - thanks for adding them.